tools.py

```diff
@@ -5,6 +5,24 @@
 
 
 class Vertex:
+    """
+    The Vertex object represents a vertex that is part of a pose graph.
+
+    Args:
+        run_id (int): Id of the run the vertex belongs to.
+        pose_id (int): Id of the pose for the vertex.
+        next_run_id (int): Id of the next consecutive run.
+        next_pose_id (int): Id of the next consecutive pose.
```

Comment:
This isn't quite right but is more because my nomenclature is confusing. Together these form the ID of the vertex this vertex has its transformation to. e.g. for (0,408) these are (0,409) but for (34,123) its (0,163). Just realised for our uses `next_run_id` should always be 0 so it's kind of redundant.

Reply:
OK, I see. I wasn't entirely sure what was meant. Would the following be correct?

next_run_id (int): Run id for the teach vertex that this vertex is connected to, either spatially (from repeat to teach) or temporally (one teach vertex to another).

next_pose_id (int): Pose id for the teach vertex that this vertex is connected to, either spatially (from repeat to teach) or temporally (one teach vertex to another).

Reply:
Yep, that's good